File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
#include <unordered_set>
using namespace std;

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> set(nums.begin(), nums.end());
        if (set.size() == nums.size())
        {
            return false;
        }
        return true;
    }
};

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        size = len(nums)
        left = 0
        right = size - 1

        while left <= right:
            mid = int((right + left)/ 2)
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                right = mid - 1
            else:
                left = mid + 1
        
        return -1